apps/console/src/features/authentication/hooks/use-sign-in.ts

```diff
@@ -449,6 +460,46 @@ const useSignIn = (): UseSignInInterface => {
         setCustomServerHost(orgType, wellKnownEndpoint);
     };
 
+    /**
+     * Derives the logout redirect URL for sub-org logins.
+     *
+     * @remarks This only applies to the new authz runtime.
+     *
+     * @param currentLogoutRedirect - Current logout redirect URL.
+     * @param userOrg - User's org.
+     * @param orgId - User's org ID.
+     * @returns Derived logout redirect URL.
+     */
+    const deriveLogoutRedirectForSubOrgLogins = (currentLogoutRedirect: string, userOrg: string, orgId: string) => {
+        if (legacyAuthzRuntime) {
+            return currentLogoutRedirect;
+        }
+
+        let logoutRedirectUrl: string = currentLogoutRedirect;
+
+        // When a first level tenant login happens, `user_org` is undefined.
+        // We need to save that in the local storage to handle the login/logout if they switch.
+        if (userOrg === undefined && orgId) {
+            removeUserOrgInLocalStorage();
+            removeOrgIdInLocalStorage();
+
+            setUserOrgInLocalStorage(userOrg);
+            setOrgIdInLocalStorage(orgId);
+        }
+
+        if (userOrg) {
+            const isSwitchedFromRootOrg: boolean = getUserOrgInLocalStorage() === "undefined";
+
+            if (!isSwitchedFromRootOrg) {
+                logoutRedirectUrl = window["AppUtils"]?.getConfig()?.clientOriginWithTenant?.replace(
```

Comment:
Shall we get this from the redux store?

Reply:
Will be fixed in a followup PR.

Reply:
Fixed in https://github.com/wso2/identity-apps/pull/5141/commits/2df86545a8bcb8d4d076e4395033ab48c1b4874c